### PR TITLE
Support/improvements

### PR DESCRIPTION
**Pull request by joobacca**
Just some type fixes and minor visual changes, check it out

### CODE REVIEW

File: app/about-us/layout.tsx [hunk 1]
```diff
@@ -32,7 +32,7 @@ export default async function AboutUsLayout({
 }) {
   const {
     fields: { homeIconComponent },
-  }: Entry<IHomeIconResuableFields> = await getHomeIcons();
+  } = await getHomeIcons();
```
Comment (diegoDeMiguelNA) [inline app/about-us/layout.tsx:35]: No Types needed here, apparently

Comment (joobacca) [reply app/about-us/layout.tsx:35]: getHomeIcons() does provide a return type, so no need to explicitly set a type

File: app/components/homePageComponents/ExperienciasDeVida.tsx [hunk 2]
```diff
@@ -1,40 +1,52 @@
...
-  const results: Entry<IBlogSubtitleParagraphAndImageFields> =
-    await getSubtitleAndParagraphAndImage("2FSjWb3HaXM9obR4vqS1Gr");
+  const results = await getSubtitleAndParagraphAndImage(
+    "2FSjWb3HaXM9obR4vqS1Gr"
+  );
 
   const { subtitle, paragraph, assets } = results.fields;
 
+  // is this check necessary?
+  if (Object.keys(results).length === 0) return;
```
Comment (diegoDeMiguelNA) [inline app/components/homePageComponents/ExperienciasDeVida.tsx:31]: I guess not, if we implement the built in error handling that NextJS offers

**Comment by joobacca** [comment]
One important thing we missed when setting the project up was using a configured formatter, I'll do that as soon as this PR is merged. We will probably have to format every single file but the code is kind of all over the place, I think it's much needed :) Especially making the format errors visible using the ESLint Prettier plugin

File: lib/contentful/apiClient.ts [hunk 3]
```diff
@@ -1,13 +1,10 @@
...
-const space = process.env.NEXT_PUBLIC_SPACE;
-const accessToken = process.env.NEXT_PUBLIC_ACCESS_TOKEN;
+export const contentfulClient = createClient({
+  accessToken: process.env.CONTENTFUL_ACCESS_TOKEN!,
+  space: process.env.CONTENTFUL_SPACE!,
+});
 
-if (!space || !accessToken) {
-  throw new Error("Space or Access Token is missing in environment variables.");
+export async function getEntityData<T>(entryId: string) {
```
Comment (diegoDeMiguelNA) [inline lib/contentful/apiClient.ts:8]: I guess this is why you removed all the "Entry<IHomeIconResuableFields>" on the layout files.

File: lib/contentful/fetchDataFromContentful.ts [hunk 4]
```diff
@@ -1,25 +1,33 @@
...
+  svgFileName?: string;
+  width?: number;
+  height?: number;
 }
+export type SubPageData = {
+  fields: SubPageFields;
+  sys: { id: string };
+};
+
+type HomeIconPageFields = Omit<IHomeIconResuableFields, "homeIconComponent"> & {
```
Comment (diegoDeMiguelNA) [inline lib/contentful/fetchDataFromContentful.ts:24]: nice, I didn't knew about "Omit"

File: utils/renderText.tsx [hunk 5]
```diff
@@ -32,7 +32,7 @@ export const RenderContent = (props: {
         if (entry.sys.contentType.sys.id === "onlyParagraph") {
           return (
             <BlogOnlyParagraph
-              key={Math.random() * index}
+              key={entry.sys.id}
```
Comment (diegoDeMiguelNA) [inline utils/renderText.tsx:35]: "sys" seems to be a more logical value to be passed as "key" than the crap I was using.

**Review by diegoDeMiguelNA — APPROVED** [review]
It was a great opportunity for me to see how to improve the code. Thanks Joo

**Pull request merged**